obspy/io/mseed/util.py

```diff
@@ -238,31 +275,110 @@ def get_timing_and_data_quality(file_or_file_object):
     info = get_record_information(file_or_file_object)
     # Keep track of the extracted information.
     quality_count = [0, 0, 0, 0, 0, 0, 0, 0]
+    activity_count = [0, 0, 0, 0, 0, 0, 0]
+    io_count = [0, 0, 0, 0, 0, 0]
```

Comment:
We do alreay have a new function here so this is a good opportunity to change the return types. I think a dictionary with flag counts would be more convenient. What do you think?

e.g.

```python
io_count = collections.OrderedDict(
    station_volume_parity_error=0,
    long_record_read=0,
    short_record_read=0, ...)
```

This is easier to read and reason about and thus should reduce usage errors in the long run. The `OrderedDict` means that `io_count.values()` is exactly the same as the current return value.